### ISSUE TITLE
Support XOR in math operations

### ISSUE THREAD

**Issue by koryphaee** [lucasteles/Strongly#17]
I am currently working on a project that uses a `ulong` as a central type in the business logic. It represents a 64bit [image hash](https://github.com/JohannesBuchner/imagehash) that I need to XOR with another hash to find the hamming distance. Currently only addition, substraction, multiplication, division, negation and comparison are supported.
I am already using this cool library for my database primary keys but can't use if for the hash yet.

From what I can tell it won't be a lot of work and I will try to create the PR myself.

I already created a separate issue for support for unsigned integers: https://github.com/lucasteles/Strongly/issues/16

**Comment by koryphaee** [comment]
I have the change ready but https://github.com/lucasteles/Strongly/pull/18 would need to be merged first since XOR is also supported for the new unsigned types.

**Comment by lucasteles** [comment]
sure!

But would not be better to have all bit operators?

**Comment by koryphaee** [comment]
Yes it would be more consistent. I will add them.
